modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -672,18 +1052,837 @@ export default class ortcRTCPeerConnection extends yaeti.EventTarget {
+            const rtpReceiver = info.rtpReceiver;
+            const track = rtpReceiver.track;
+
+            try {
+                rtpReceiver.stop();
+            } catch (error) {
+                logger.warn(`rtpReceiver.stop() failed:${error}`);
+            }
+
+            removedRemoteTracks.set(track, stream);
+            stream.removeTrack(track);
+            this._remoteTrackInfos.delete(ssrc);
+        }
+
+        // Emit MediaStream 'addtrack' for new tracks in already existing
+        // streams.
+        for (const pair of addedRemoteTracks) {
+            const track = pair[0];
+            const stream = pair[1];
+            const event = new Event('addtrack');
+
+            event.track = track;
+            stream.dispatchEvent(event);
+        }
+
+        // Emit MediaStream 'removetrack' for removed tracks.
+        for (const pair of removedRemoteTracks) {
+            const track = pair[0];
+            const stream = pair[1];
+            const event = new Event('removetrack');
+
+            event.track = track;
+            stream.dispatchEvent(event);
+        }
+
+        // Emit RTCPeerConnection 'addstream' for new remote streams.
+        for (const stream of addedRemoteStreams.values()) {
+            this._emitAddStream(stream);
+        }
+
+        // Emit RTCPeerConnection 'removestream' for removed remote streams.
+        for (const pair of this._remoteStreams) {
+            const streamRemoteId = pair[0];
+            const stream = pair[1];
+
+            if (stream.getTracks().length > 0) {
+                continue; // eslint-disable-line no-continue
+            }
+
+            this._remoteStreams.delete(streamRemoteId);
+            this._emitRemoveStream(stream);
+        }
+    }
+
+    /**
+     * Get RTP parameters for a RTCRtpSender.
+     * @private
+     * @return {RTCRtpParameters}
+     */
+    _getParametersForRtpSender(kind, data) {
+        const ssrc = data.ssrc;
+        const rtxSsrc = data.rtxSsrc;
+        const cname = CNAME;
+        const localCapabilities = this._localCapabilities;
+        const parameters = {
+            muxId: '',
+            degradationPreference: 'balanced',
+            codecs: [],
+            headerExtensions: [],
+            encodings: [],
+            rtcp: {
+                cname,
+                reducedSize: true, // NOTE: Not yet implemented in Edge.
+                compound: true, // NOTE: Implemented in Edge.
+                mux: true
+            }
+        };
+
+        const codecs = [];
+        let codecPayloadType;
+
+        for (const codecCapability of localCapabilities.codecs) {
+            if (codecCapability.kind !== kind
+                || codecCapability.name === 'rtx') {
+                continue; // eslint-disable-line no-continue
+            }
+
+            codecPayloadType = codecCapability.preferredPayloadType;
+            codecs.push({
+                name: codecCapability.name,
```

Comment:
No please, keys are put in the same order as they appear in the spec. `payloadType` is MUCH more important than `clockRate`, etc.